### ISSUE TITLE
partial-update, support update `Builder`'s implemented types

### ISSUE THREAD

**Issue by haolingdong-msft** [Azure/autorest.java#2420]

**Comment by weidongxu-microsoft** [comment]
Yeah, we may want to limit this extra processing to the client builder class (not sure how to narrow it though, at least we can check `##Builder`, or exist of `@ServiceClientBuilder` annotation but not sure if it still there in generic core).
